src/topology/algebra/uniform_group.lean

```diff
@@ -200,37 +200,81 @@ def topological_group.to_uniform_space : uniform_space G :=
 
-@[to_additive] lemma topological_group.tendsto_uniformly_on_iff
-  {ι α : Type*} (F : ι → α → G) (f : α → G) (p : filter ι) (s : set α) :
-  @tendsto_uniformly_on α G ι (topological_group.to_uniform_space G) F f p s
-    ↔ ∀ u ∈ 𝓝 (1 : G), ∀ᶠ i in p, ∀ a ∈ s, F i a / f a ∈ u :=
+@[to_additive] lemma tendsto_uniformly_on_iff :
+  tendsto_uniformly_on F f p s ↔ ∀ u ∈ 𝓝 (1 : G), ∀ᶠ i in p, ∀ a ∈ s, F i a / f a ∈ u :=
 ⟨λ h u hu, h _ ⟨u, hu, λ _, id⟩, λ h v ⟨u, hu, hv⟩,
   mem_of_superset (h u hu) (λ i hi a ha, hv (by exact hi a ha))⟩
 
-@[to_additive] lemma topological_group.tendsto_locally_uniformly_iff
-  {ι α : Type*} [topological_space α] (F : ι → α → G) (f : α → G) (p : filter ι) :
-  @tendsto_locally_uniformly α G ι (topological_group.to_uniform_space G) _ F f p
+@[to_additive] lemma tendsto_locally_uniformly_iff [topological_space α] :
+  tendsto_locally_uniformly F f p
     ↔ ∀ (u ∈ 𝓝 (1 : G)) (x : α), ∃ (t ∈ 𝓝 x), ∀ᶠ i in p, ∀ a ∈ t, F i a / f a ∈ u :=
 ⟨λ h u hu, h _ ⟨u, hu, λ _, id⟩, λ h v ⟨u, hu, hv⟩ x, exists_imp_exists (by exact λ a,
   exists_imp_exists (λ ha hp, mem_of_superset hp (λ i hi a ha, hv (by exact hi a ha)))) (h u hu x)⟩
 
-@[to_additive] lemma topological_group.tendsto_locally_uniformly_on_iff
-  {ι α : Type*} [topological_space α] (F : ι → α → G) (f : α → G) (p : filter ι) (s : set α) :
-  @tendsto_locally_uniformly_on α G ι (topological_group.to_uniform_space G) _ F f p s
+@[to_additive] lemma tendsto_locally_uniformly_on_iff [topological_space α] :
+  tendsto_locally_uniformly_on F f p s
     ↔ ∀ (u ∈ 𝓝 (1 : G)) (x ∈ s), ∃ (t ∈ 𝓝[s] x), ∀ᶠ i in p, ∀ a ∈ t, F i a / f a ∈ u :=
 ⟨λ h u hu, h _ ⟨u, hu, λ _, id⟩, λ h v ⟨u, hu, hv⟩ x, exists_imp_exists (by exact λ a,
   exists_imp_exists (λ ha hp, mem_of_superset hp (λ i hi a ha, hv (by exact hi a ha)))) ∘ h u hu x⟩
 
-end topological_comm_group
+@[to_additive] lemma uniform_continuous_mul : uniform_continuous₂ (has_mul.mul : G → G → G) :=
+begin
+  rintros v ⟨u, hu, huv⟩,
+  obtain ⟨w, hw, x, hx, h⟩ := mem_nhds_prod_iff.mp (mem_map.mp
+    (continuous_mul.tendsto' ((1 : G), (1 : G)) (1 : G) (one_mul (1 : G)) hu)),
+  rw [uniformity_prod_eq_prod, mem_map, mem_map, mem_prod_iff],
+  refine ⟨_, preimage_mem_comap hw, _, preimage_mem_comap hx, _⟩,
+  rw [set.prod_preimage_eq],
+  exact λ a ha, huv (by exact ((congr_arg (∈ u) (div_mul_comm _ _ _ _)).mp (h ha))),
+end
+
+@[to_additive] lemma uniform_continuous_inv : uniform_continuous (has_inv.inv : G → G) :=
+begin
+  rintros v ⟨u, hu, huv⟩,
+  refine ⟨_, continuous_inv.tendsto' _ _ one_inv hu, λ g hg, huv ((congr_arg (∈ u) _).mp hg)⟩,
+  exact (inv_div' g.2 g.1).trans (inv_div_inv g.2 g.1).symm,
+end
+
+@[to_additive] lemma tendsto_uniformly_on_mul
+  (h₁ : tendsto_uniformly_on F₁ f₁ p₁ s) (h₂ : tendsto_uniformly_on F₂ f₂ p₂ s) :
+  tendsto_uniformly_on (λ i : ι₁ × ι₂, F₁ i.1 * F₂ i.2) (f₁ * f₂) (p₁.prod p₂) s :=
+begin
+  letI : uniform_space (G × G) := prod.uniform_space,
```

Comment:
```suggestion
  -- Diamond: `to_uniform_space (G × G)` not defeq to `prod.uniform_space`
  letI : uniform_space (G × G) := prod.uniform_space,
```

Reply:
I've raised this [on Zulip](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.2311470.20Uniform.20convergence.20of.20multiplication.20and.20inversion) so I'll give people some time to answer before making a call.

Given that you're only adding `lemma`s here I think this is fine to merge: the worst that could happen is somebody might find `tendsto_uniformly_on_mul` or `tendsto_uniformly_mul` might not apply in some situation (until we resolve this diamond issue).